app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
Fair enough. I didn't have a lot of context around this, so it just seemed a bit odd to me on first glance.